rclcpp/include/rclcpp/node_interfaces/node_base.hpp

```diff
@@ -104,8 +104,8 @@ class NodeBase : public NodeBaseInterface
   get_associated_with_executor_atomic() override;
 
   RCLCPP_PUBLIC
-  rcl_guard_condition_t *
-  get_notify_guard_condition() override;
+  rclcpp::GuardCondition::SharedPtr
+  get_notify_guard_condition() const override;
```

Comment:
If you add `const` the return type should be `const` too, either `shared_ptr<const rclcpp::GuardConditioon>` or, as I suggested, `const rclcpp::GuardCondition &`.